app/views/reports/donationsReport.html

```diff
@@ -3,9 +3,37 @@
   <span class="h3 page-header">Donations report</span>
   
   <hr style="margin-top:5px;"/>
+  
+  <form class="form-horizontal" role="form"  name="selectPeriodForm" style="padding:10px 0;" ng-submit="getReport(selectPeriodForm)" novalidate>
+    <div class="clearfix form-group">
+        <label class="col-sm-2 control-label">Period:</label>
+        <div class="form-inline col-sm-10">
+            <div>
+                <dateselect name="startDate" ng-model="search.startDate" format="dateFormat" ng-class="{'has-error' : (selectPeriodForm.startDate.$invalid)}" ng-required="true" ui-date-range="1,months" ui-date-end="{{search.endDate | date:'yyyy-MM-dd HH:mm:ss'}}" ng-model-options="{ allowInvalid: true }"></dateselect>
+                <span><em>To:</em></span>
+                <dateselect-end name="endDate" ng-model="search.endDate" format="dateFormat" ng-class="{'has-error' : (selectPeriodForm.endDate.$invalid)}" ng-required="true" ui-date-range="1,months" ui-date-start="{{search.startDate |  date:'yyyy-MM-dd HH:mm:ss'}}" ng-model-options="{ allowInvalid: true }"></dateselect-end>
+            </div>
+            <div>
+                <small class="error" ng-show="(selectPeriodForm.startDate.$error.required || selectPeriodForm.endDate.$error.required)">
+                    Select a valid date
+                </small>
+                <small class="error" ng-show="(selectPeriodForm.startDate.$error.uiDateRange || selectPeriodForm.endDate.$error.uiDateRange)">
+                    Select a date range that is less than one month
+                </small>
+            </div>
+        </div>
+      </div>
+
+      <div class="col-sm-offset-2 ">
+          <button class="btn btn-primary" type="submit" ng-disabled="searching">{{searching ? "Searching..." :
+              "Search"}}
+          </button>
+          <button class="btn btn-primary" type="reset" ng-click="clearSearch(selectPeriodForm)">Clear</button>
```

Comment:
To fix the problem with clearing the form (that the "master" start/end dates aren't persisted, replace `type="reset"` with `type="button"`. What was happening is that the default reset functionality was clearing the values in the form fields (after the default was restored).